plugin/storage/s3/s3.go

```diff
@@ -12,23 +13,32 @@ import (
 	"github.com/pkg/errors"
 )
 
-// Credentials is the AWS S3 credentials.
-type Credentials struct {
-	AccessKeyID     string `json:"accessKeyId"`
-	SecretAccessKey string `json:"secretAccessKey"`
-}
-
 // Client wraps the AWS S3 client.
 type Client struct {
 	c      *s3.Client
 	bucket string
 }
 
+// GetCredentialsFromFile load AWS credentials from file.
+func GetCredentialsFromFile(ctx context.Context, credentialsFileName string) (aws.Credentials, error) {
+	cfg, err := awsconfig.LoadDefaultConfig(ctx,
+		awsconfig.WithSharedCredentialsFiles([]string{credentialsFileName}),
+	)
+	if err != nil {
+		return aws.Credentials{}, errors.Wrap(err, "failed to load AWS S3 config")
+	}
+	credentials, err := cfg.Credentials.Retrieve(ctx)
+	if err != nil {
+		return aws.Credentials{}, errors.Wrap(err, "failed to retrieve AWS S3 credentials")
+	}
+	return credentials, nil
+}
+
 // NewClient returns a new AWS S3 client.
-func NewClient(ctx context.Context, region, bucket string, credentials Credentials) (*Client, error) {
+func NewClient(ctx context.Context, region, bucket string, credentials aws.Credentials) (*Client, error) {
 	cfg, err := awsconfig.LoadDefaultConfig(ctx,
 		awsconfig.WithRegion(region),
-		awsconfig.WithCredentialsProvider(awscredentials.NewStaticCredentialsProvider(credentials.AccessKeyID, credentials.SecretAccessKey, "")),
+		awsconfig.WithCredentialsProvider(awscredentials.NewStaticCredentialsProvider(credentials.AccessKeyID, credentials.SecretAccessKey, credentials.SessionToken)),
```

Comment:
What's the lifetime of SessionToken?

Reply:
It depends on the AWS STS.
Second thought: this makes the solution complicated. I'll remove the support for session token for now.

Reply:
What's the AI?

Reply:
Changed back to `""`. We do not support session token for now.